cmd/initialize_with_genesis.cpp

```diff
@@ -0,0 +1,335 @@
+
+                // Make the account
+                account_changeset_table->put(block_number, *address_bytes);
+                plainstate_table->put(*address_bytes, account.encode_for_storage(true));
+
+                // Fills hash builder
+                auto hash{keccak256(*address_bytes)};
+                account_rlp[to_bytes32(hash.bytes)] = account.rlp(kEmptyRoot);
+            }
+
+            auto it{account_rlp.cbegin()};
+            trie::HashBuilder hb{full_view(it->first), it->second};
+            for (++it; it != account_rlp.cend(); ++it) {
+                hb.add(full_view(it->first), it->second);
+            }
+            root_hash = hb.root_hash();
+
+        } else {
+            root_hash = kEmptyRoot;
+        }
+
+        // Fill Header
+        BlockHeader header;
+        header.ommers_hash = kNullOmmers;
+        header.state_root = root_hash;
+        header.transactions_root = kEmptyRoot;
+        header.receipts_root = kEmptyRoot;
+
+        auto difficulty_str{genesis_json["difficulty"].get<std::string>()};
+        header.difficulty = intx::from_string<intx::uint256>(difficulty_str);
+
+        header.gas_limit = std::stoull(genesis_json["gasLimit"].get<std::string>().c_str(), nullptr, 0);
+        header.timestamp = std::stoull(genesis_json["timestamp"].get<std::string>().c_str(), nullptr, 0);
+
+        auto nonce_str{genesis_json["nonce"].get<std::string>()};
+        auto nonce_bytes{from_hex(nonce_str)};
+        auto diff_nonce_size(8 - nonce_bytes->size());
+        for (size_t i = 0; i < nonce_bytes->size(); i++)
+            header.nonce[i + diff_nonce_size] = nonce_bytes->at(i + diff_nonce_size);
+
+        // Write header
+        auto blockhash{header.hash()};
+
+        Bytes rlp_header;
+        rlp::encode(rlp_header, header);
+        Bytes key(8 + kHashLength, '\0');
+        std::memcpy(&key[8], blockhash.bytes, kHashLength);
+        txn->open(db::table::kHeaders)->put(key, rlp_header);
+        txn->open(db::table::kCanonicalHashes)->put(block_number, full_view(blockhash.bytes));
+        // Write body
+        txn->open(db::table::kBlockBodies)->put(key, Bytes(genesis_body, 4));
+        txn->open(db::table::kDifficulty)->put(key, intx::as_bytes(header.difficulty));
+        txn->open(db::table::kBlockReceipts)->put(key.substr(0, 8), Bytes(genesis_receipts, 1));
+        txn->open(db::table::kHeadHeader)
+            ->put(Bytes(reinterpret_cast<const uint8_t*>(last_header_key.c_str()), last_header_key.size()),
+                  full_view(blockhash.bytes));
+        txn->open(db::table::kHeaderNumbers)->put(full_view(blockhash.bytes), key.substr(0, 8));
+        // Write Chain Config
+        auto chain_config{genesis_json["config"].dump()};
+        txn->open(db::table::kConfig)
+            ->put(full_view(blockhash.bytes),
+                  Bytes(reinterpret_cast<const uint8_t*>(chain_config.c_str()), chain_config.size()));
```

Comment:
I would extract some functions into `silkworm/db` and add a test that
1. initializes a temp DB
2. reads the config with `parse_chain_config`
3. checks that the config equals kMainnetConfig, kRinkebyConfig, etc

That will ensure that `core/silkworm/chain/config.hpp` is in line with the JSON genesis files.